src/Bicep.LangServer/BicepParamsCompilationManager.cs

```diff
@@ -35,8 +48,17 @@ public void RefreshCompilation(DocumentUri uri, bool reloadBicepConfig = false)
 
         public void UpsertCompilation(DocumentUri uri, int? version, string text, string? languageId = null)
         {
-            var semanticModel = new ParamsSemanticModel(SourceFileFactory.CreateBicepParamFile(uri.ToUri(), text));
+            var paramsFile = SourceFileFactory.CreateBicepParamFile(uri.ToUri(), text);
+            var semanticModel = new ParamsSemanticModel(paramsFile);
+            var usingDeclarations = paramsFile.ProgramSyntax.Children.OfType<UsingDeclarationSyntax>();
 
+            if (PathHelper.TryGetUsingPath(usingDeclarations.SingleOrDefault(), out var bicepFilePath, out var _) && Uri.TryCreate(uri.ToUri(), bicepFilePath, out var bicepFileUri))
```

Comment:
Could we move this logic into the `ParamsSemanticModel` so that it can be shared between language server & CLI?

Reply:
I was thinking the logic to:
* Find the using statement, raise diagnostics if it missing or duplicated
* Read the URI, ensure it is valid, raise diagnostics if not
* Return the complete expected URI

Feel free to address this in your next PR (as per your other comment on missing diags)

Generating the compilation itself is probably going to be problematic, since it's going to differ between CLI & LangServer.